File: sdks/go/user_prompts.go
// Copyright (C) 2026 Michael Kao
// SPDX-License-Identifier: AGPL-3.0-or-later

package llmsafespaces

import (
	"context"
)

// User prompt library (#1499) — the caller's saved prompts
// (/me/prompts). Owner-scoped CRUD; the composer @-recall surface
// reads the same list. Distinct from the platform prompt-POLICY
// surface (platform/org/workspace prompts).

// UserPromptsService manages the caller's saved prompts.
type UserPromptsService struct{ c *Client }

func (s *UserPromptsService) List(ctx context.Context) ([]UserPrompt, error) {
	var resp struct {
		Prompts []UserPrompt `json:"prompts"`
	}
	if err := s.c.do(ctx, "GET", "/me/prompts", nil, &resp); err != nil {
		return nil, err
	}
	return resp.Prompts, nil
}

func (s *UserPromptsService) Create(ctx context.Context, req CreateUserPromptRequest) (*UserPrompt, error) {
	var resp struct {
		Prompt UserPrompt `json:"prompt"`
	}
	if err := s.c.do(ctx, "POST", "/me/prompts", req, &resp); err != nil {
		return nil, err
	}
	return &resp.Prompt, nil
}

func (s *UserPromptsService) Update(ctx context.Context, id string, req UpdateUserPromptRequest) (*UserPrompt, error) {
	var resp struct {
		Prompt UserPrompt `json:"prompt"`
	}
	if err := s.c.do(ctx, "PUT", "/me/prompts/"+id, req, &resp); err != nil {
		return nil, err
	}
	return &resp.Prompt, nil
}

func (s *UserPromptsService) Delete(ctx context.Context, id string) error {
	return s.c.do(ctx, "DELETE", "/me/prompts/"+id, nil, nil)
}

File: frontend/src/api/userPrompts.ts
// Copyright (C) 2026 Michael Kao
// SPDX-License-Identifier: AGPL-3.0-or-later

import { api } from "./client";

// #1499: the user-prompt library adapter (named distinctly from
// prompts.ts — the platform/org prompt-POLICY surface, a different
// concept that owns the bare "prompts" noun here). The composer
// #-recall lane (#1496) consumes the same list contract via its own
// adapter.
export interface UserPrompt {
  id: string;
  name: string;
  content: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreatePromptRequest {
  name: string;
  content: string;
}

export interface UpdatePromptRequest {
  name?: string;
  content?: string;
}

export const userPromptsApi = {
  list: () => api.get<{ prompts: UserPrompt[] }>("/me/prompts"),
  create: (req: CreatePromptRequest) => api.post<{ prompt: UserPrompt }>("/me/prompts", req),
  update: (id: string, req: UpdatePromptRequest) => api.put<{ prompt: UserPrompt }>(`/me/prompts/${id}`, req),
  delete: (id: string) => api.delete<void>(`/me/prompts/${id}`),
};
